deps/jerry/jerry-core/ecma/base/ecma-alloc.c

```diff
@@ -28,8 +28,10 @@ JERRY_STATIC_ASSERT (((sizeof (ecma_property_value_t) - 1) & sizeof (ecma_proper
 JERRY_STATIC_ASSERT (sizeof (ecma_string_t) == sizeof (uint64_t),
                      size_of_ecma_string_t_must_be_less_than_or_equal_to_8_bytes);
 
+#ifndef JERRY_HEAP_PROFILER
 JERRY_STATIC_ASSERT (sizeof (ecma_extended_object_t) - sizeof (ecma_object_t) <= sizeof (uint64_t),
                      size_of_ecma_extended_object_part_must_be_less_than_or_equal_to_8_bytes);
+#endif /* JERRY_HEAP_PROFILER */
```

Comment:
Why do we disable this assertion in heap profiling?

Reply:
Before this PR, sizeof (ecma_extended_object_t) - sizeof (ecma_object_t) == sizeof (uint64_t).
But I add a size field into ecma_extended_object_t, which make the assertion failed. This field will increase 4 bytes for every ext object. To save memory without profiler, I would wrap it by macro.

Reply:
> This field will increase 4 bytes for every ext object.

How about still doing assertion but like this:

```c
#ifndef JERRY_HEAP_PROFILER
JERRY_STATIC_ASSERT (sizeof (ecma_extended_object_t) - sizeof (ecma_object_t) <= sizeof (uint64_t),
                     size_of_ecma_extended_object_part_must_be_less_than_or_equal_to_8_bytes);
#else /* JERRY_HEAP_PROFILER */
JERRY_STATIC_ASSERT (sizeof (ecma_extended_object_t) - sizeof (ecma_object_t) - sizeof (EXT_OBJ_SIZE_TYPE) <= sizeof (uint64_t),
                     size_of_ecma_extended_object_part_must_be_less_than_or_equal_to_8_bytes);
#endif
```